Feature(JoinReactTest): Daniel Morozov

hameltomor opened the pull request:
### Info ℹ️ 

Didn't complete full requirements :(


### Preview version

Deployed to [vercel](https://join-react-test.vercel.app/).


### Screenshots 🖼️ 

![image](https://user-images.githubusercontent.com/6232083/90389259-e11ee700-e091-11ea-8f20-99df207e27b6.png)

![image](https://user-images.githubusercontent.com/6232083/90389416-38bd5280-e092-11ea-8d14-2f9c54965605.png)

michowski commented on src/components/Form/Checkbox/index.tsx:
```diff
@@ -0,0 +1,86 @@
+	formikValue: any;
+	label?: string;
+
+	onChange?: any;
```
Could you find more suitable types for these `any`s?

hameltomor replied:
Yes, you are right, I have a more correct description of this type. I will change it

michowski commented on src/components/Layout/Footer/parts/Link/index.tsx:
```diff
@@ -0,0 +1,7 @@
+import styled from 'styled-components'
+
+const Link = styled.a`
```
Are these "empty" multiline strings necessary?

hameltomor replied:
Removed

michowski commented on src/components/Partials/CandidateItem/index.tsx:
```diff
@@ -0,0 +1,93 @@
+
+	const onDropdownChange = async (selectedItem: any) => {
+		let state: CandidateState
+		switch (Number(selectedItem.key)) {
```
Do you see a way to greatly simplify this entire conversion from number to `CandidateState` values?

hameltomor replied:
As a `key` in the dropdown can be a value from CandidateState. It allows to skip `switch` and just set a new value to candidate

hameltomor replied:
Done

michowski commented on src/components/Partials/CandidateItem/parts/Dropdown/index.tsx:
```diff
@@ -0,0 +1,32 @@
+			onSelect={handleOnSelect}
+		>
+			<SubMenu title=". . .">
+				<MenuItem key="1">Delete</MenuItem>
```
Again, this mapping from string values to "hard-coded" numbers can become problematic. Do you see a potential problem in maintaining it over time?

Do you see a way to simplify it?

hameltomor replied:
Yep, use the keys based on the values by Enum `CandidateState`

hameltomor replied:
Done

michowski commented on src/services/api/candidates.ts:
```diff
@@ -0,0 +1,50 @@
+});
+
+export const getAll = async (): Promise<Candidate[]> => {
+	const { data: candidates = [] } = await API.get(`/candidates`) as ApiResponse<Candidate[]>;
```
Is there a reason for this default value here? (`[]`)

hameltomor replied:
Yes, because components get this data from their props without a typescript symbol `?`, so it means that this fields are required and we must pass 100% not undefined value. It allows to skip a checking condition into component

michowski commented on src/stories/Header.tsx:
```diff
@@ -0,0 +1,47 @@
+import './header.css';
+
+export interface HeaderProps {
+  user?: {};
```
is this a proper type for the `user`?

hameltomor replied:
There are default stories by storybook, did not have time to change them :(

hameltomor replied:
Done

hameltomor commented:
@michowski Hey 🖐️ 

Thanks for review!

> Would you like to spend some additional time and work on things that you consider to be potentially improved?

Of course I want to improve. I have not done a lot of things that I wanted to implement :)
I will provide changes based on comments.

Have a nice day,
Daniel

hameltomor commented:
### Info ℹ️ 

The 01205515983f367b200d5d2a9547c03aa3bd83fd commit contains fixes which contains covers all comments above.


### Changes 🗒️ 

- Fixed typings (code without `any`)
- Actions for candidate applications in the menu based on the `Enum`
- Styled forms
- Added SEO tags - title, description


### Screenshots 🖼️ 

- Home page
![image](https://user-images.githubusercontent.com/6232083/90804723-4806ff00-e323-11ea-953f-f99c18e77c90.png)

- Recruiter page
![image](https://user-images.githubusercontent.com/6232083/90804764-5523ee00-e323-11ea-92f2-874c760b31a2.png)

- Candidate page
![image](https://user-images.githubusercontent.com/6232083/90804785-5f45ec80-e323-11ea-8369-ab86347358a1.png)
